neuralogic/core/result.py

```diff
@@ -0,0 +1,32 @@
+import json
+
+
+class Result:
+    __slots__ = "_result", "_sample", "_model", "_number_format"
+
+    def __init__(self, result, sample, model, number_format):
+        self._result = result
+        self._sample = sample
+        self._model = model
+        self._number_format = number_format
+
+    def backward(self):
+        weight_updater = self._model._trainer.backpropSample(self._model._backpropagation, self._result, self._sample)
+        self._model._trainer.updateWeights(self._model._strategy.getCurrentModel(), weight_updater)
```

Comment:
`trainer` and `backpropagation` should be fields of `Result`, unless we update those when template changes

Reply:
Result removed, irrelevant now